### PR TITLE
Created a email system- Adniya Yousaf

### PR DESCRIPTION
**Pull request by adniyaYousaf**
## Description
I have created email system using Nodemailer which is a module for Node.js applications to allow easy as cake email sending.
I have also create an endpoint with POST method to send mail.

### CODE REVIEW

File: example.env [hunk 1]
```diff
@@ -0,0 +1,7 @@
+DATABASE_URL=postgres://postgres:opensesame@localhost:5432/cyf
+
+MAIL_HOST= sandbox.smtp.mailtrap.io
+MAIL_PORT= 465
+MAIL_USER= 14e06e26e6e95c
+MAIL_PASSWORD= e93f4808aeb665
```
Comment (soonick) [inline example.env]: These are secrets, don't commit them in the code, let's add them to the bitwarded vault I created and then use environment variables.

We will probably need to update the password since this one is already on the internet.

File: example.env [hunk 2]
```diff
@@ -0,0 +1,7 @@
+DATABASE_URL=postgres://postgres:opensesame@localhost:5432/cyf
+
```
Comment (soonick) [inline example.env]: I think this doesn't belong in this PR

File: server/api.js [hunk 3]
```diff
@@ -1,9 +1,26 @@
...
+	const message = "Your passwors is ******.";
+	try {
+		await mail.sendEmail({ recipient, subject, message });
+		res.send("Email sent.");
+	} catch (error) {
+		logger.error("Error sending email:", error);
+		res.status(500).send("Failed to send email.");
+	}
+});
+
```
Comment (soonick) [inline server/api.js]: Delete this. We don't need a `send-email` endpoint

File: server/mail.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        from: "no_reply@example.com",
+        to: recipient,
+        subject,
+        text: message,
+        html: message,
+    });
+};
+
+
+
```
Comment (soonick) [inline server/mail.js]: Leave only one blank line

File: server/mail.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+/* eslint-disable no-console */
```
Comment (soonick) [inline server/mail.js]: Remove

File: server/mail.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+import nodemailer from "nodemailer";
+require("dotenv").config();
+const transporter = nodemailer.createTransport({
```
Comment (soonick) [inline server/mail.js:4]: Empty line before this line

File: server/mail.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    host: process.env.MAIL_HOST,
+    port: process.env.MAIL_PORT,
+    secure: false,
```
Comment (soonick) [inline server/mail.js]: What does this mean?

File: server/mail.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    auth: {
+        user: process.env.MAIL_USER,
+        pass: process.env.MAIL_PASSWORD,
+    },
+});
+
+const sendEmail = async ({ recipient, subject, message }) => {
+    return await transporter.sendMail({
+        from: "no_reply@example.com",
```
Comment (soonick) [inline server/mail.js:16]: Were you able to send and receive emails successfully?, this looks very spammy

File: example.env [hunk 9]
```diff
@@ -0,0 +1,7 @@
...
+MAIL_HOST= sandbox.smtp.mailtrap.io
+MAIL_PORT= 465
+MAIL_USER= 14e06e26e6e95c
+MAIL_PASSWORD= e93f4808aeb665
+
```
Comment (soonick) [inline example.env]: Delete empty line

**Review by soonick — CHANGES_REQUESTED** [review]

File: server/mail.js (hunk 7 above)
Comment (adniyaYousaf) [reply server/mail.js]: It means weather you want to use SSL(Secure Sockets Layer) for the email or not.

File: server/mail.js (hunk 8 above)
Comment (adniyaYousaf) [reply server/mail.js:16]: Yes! I use the [Mailtrap](https://mailtrap.io/home) to make sure it send and receive emails successfully. Just added its credentials.

File: server/mail.js (hunk 7 above)
Comment (soonick) [reply server/mail.js]: Why did you choose not to use SSL?

File: server/mail.js (hunk 8 above)
Comment (soonick) [reply server/mail.js:16]: That's strange. You will most likely get blocked if you send email from a domain you don't own (example.com), but if it works, we can leave it as is for now

File: example.env [hunk 10]
```diff
@@ -0,0 +1,4 @@
+MAIL_HOST= sandbox.smtp.mailtrap.io
+MAIL_PORT= 465
+MAIL_USER= 14e06e26e6e95c
+MAIL_PASSWORD= e93f4808aeb665
```
Comment (soonick) [inline example.env]: In case my comment wasn't clear. We should not have credentials committed in the code

**Review by soonick — APPROVED** [review]
